apollo-federation/src/operation/tests/mod.rs

```diff
@@ -2031,6 +1941,7 @@ fn reuse_fragments_with_non_intersecting_types() {
         }
     "#;
 
+    // dropping unnecessary __typename selection from fragment
```

Comment:
I think the correctness checker needs to understand when `__typename` should be dropped. I'll follow-up on this.